extensions/arc/src/ui/components/filePicker.ts

```diff
@@ -0,0 +1,100 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as azdata from 'azdata';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as loc from '../../localizedConstants';
+import { IReadOnly } from '../dialogs/connectControllerDialog';
+
+export interface RadioOptionsInfo {
+	values?: string[],
+	defaultValue: string
+}
+
+export class FilePicker implements IReadOnly {
+	private _flexContainer: azdata.FlexContainer;
+	private _filePathInputBox: azdata.InputBoxComponent;
+	private _filePickerButton: azdata.ButtonComponent;
+	constructor(
+		modelBuilder: azdata.ModelBuilder,
+		initialPath: string, onNewDisposableCreated: (disposable: vscode.Disposable) => void
+	) {
+		const buttonWidth = 80;
+		this._filePathInputBox = modelBuilder.inputBox()
+			.withProperties<azdata.InputBoxProperties>({
+				value: initialPath,
+				width: 350
+			}).component();
+
+		this._filePickerButton = modelBuilder.button()
+			.withProperties<azdata.ButtonProperties>({
+				label: loc.browse,
+				width: buttonWidth
+			}).component();
+		onNewDisposableCreated(this._filePickerButton.onDidClick(async () => {
+			const fileUris = await vscode.window.showOpenDialog({
+				canSelectFiles: true,
+				canSelectFolders: false,
+				canSelectMany: false,
+				defaultUri: this._filePathInputBox.value ? vscode.Uri.file(path.dirname(this._filePathInputBox.value)) : undefined,
+				openLabel: loc.select,
+				filters: undefined /* file type filters */
+			});
+
+			if (!fileUris || fileUris.length === 0) {
+				return; // TODO: Should we throw here. It is unclear whey showOpenDialog above will return undefined or an empty array
```

Comment:
I would expect this to happen if the user cancels out. I don't think we should throw in any case, the user just won't be able to move on until they select something.

Reply:
updated comment